Introduce debug and trace log levels in c10d

cbalioglu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #73149
* __->__ #73167
* #73166
* #73144

This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

cbalioglu committed:
Introduce debug and trace log levels in c10d

This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM, can we have a couple comments/documentation or perhaps an internal post with details on what this enables/guide on how to use this? 

For example, what's the difference between C10D_TRACE and C10D_DEBUG and when should we use one over another. And in general, should distributed developer avoid using `LOG(INFO)` in favor of these newly introduced c10d macros?

cbalioglu commented:
> LGTM, can we have a couple comments/documentation or perhaps an internal post with details on what this enables/guide on how to use this?
> 
> For example, what's the difference between C10D_TRACE and C10D_DEBUG and when should we use one over another. And in general, should distributed developer avoid using `LOG(INFO)` in favor of these newly introduced c10d macros?

Yes, I plan to update our official documentation once we launch these changes.

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]

cbalioglu committed:
Update on "Introduce debug and trace log levels in c10d"


This PR adds `C10D_DEBUG` and `C10D_TRACE` macros to enable fine grained logging in c10d. It also updates some log statements of `socket` to make its output less noisy.

Differential Revision: [D34371426](https://our.internmc.facebook.com/intern/diff/D34371426/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D34371426/)!

[ghstack-poisoned]
